sky/utils/atomic.py

```diff
@@ -0,0 +1,77 @@
+"""Atomic structures and utilties."""
+
+import threading
+
+
+class Int:
+    """A thread-safe atomic integer implementation."""
+
+    def __init__(self, initial_value: int = 0):
+        self._value = initial_value
+        self._lock = threading.Lock()
+
+    def get(self) -> int:
+        """Get the current value atomically.
+
+        Returns:
+            The current integer value.
+        """
+        with self._lock:
+            return self._value
+
+    def set(self, value: int) -> None:
+        """Set the value atomically.
+
+        Args:
+            value: The new integer value to set.
+        """
+        with self._lock:
+            self._value = value
+
+    def increment(self, delta: int = 1) -> int:
```

Comment:
minor nit: can override `__iadd__`, not feeling strongly. the current way is fine.